gquantlab/src/validator.ts

```diff
@@ -19,6 +19,56 @@ function valid(required: any, outputs: any): boolean {
   return true;
 }
 
+function validNmType(required: any, outputs: any): boolean {
```

Comment:
Add a TODO. Think about how to refactor this. Maybe adding an API for dynamically loading custom typescript modules. Refer to my previous comment.

Reply:
Yes. basically we need to build a plugin file format that includes Python node files, UI display for ports, and port compatiblitiy checks.